tests/test_filesystem.py

```diff
@@ -0,0 +1,434 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy of
+# the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+# ==============================================================================
+
+import os
+import posixpath
+import sys
+import time
+from urllib.parse import urlparse
+
+import boto3
+import pytest
+import tensorflow as tf
+import tensorflow_io as tfio  # pylint: disable=unused-import
+from azure.storage.blob import ContainerClient
+
+pytestmark = pytest.mark.skipif(
+    sys.platform in ("win32", "darwin"),
```

Comment:
Wondering if we can enable macOS/Darwin?

Reply:
Actually, looks like this part is not needed? Maybe we can remove ln 28-31?

Reply:
I will send a follow up PR for this and some other improvements.

Reply:
@vnvo2409 Sounds good!